extensions/amp-pan-zoom/0.1/amp-pan-zoom.js

```diff
@@ -517,64 +517,94 @@ export class AmpPanZoom extends AMP.BaseElement {
       }
     });
 
-    this.gestures_.onGesture(PinchRecognizer, e => {
-      const {
-        centerClientX,
-        centerClientY,
-        deltaX,
-        deltaY,
-        dir,
-        last,
-      } = e.data;
-
-      this.onPinchZoom_(centerClientX, centerClientY, deltaX, deltaY, dir);
-      if (last) {
-        this.onZoomRelease_();
-      }
-    });
+    this.gestures_.onGesture(PinchRecognizer,
+        this.handlePinch.bind(this));
 
     if (!this.disableDoubleTap_) {
-      this.gestures_.onGesture(DoubletapRecognizer, e => {
-        const {clientX, clientY} = e.data;
-        this.onDoubletapZoom_(clientX, clientY)
-            .then(() => this.onZoomRelease_());
-      });
+      this.gestures_.onGesture(DoubletapRecognizer,
+          this.handleDoubleTap.bind(this));
       // Override all taps to enable tap events on content
-      this.gestures_.onGesture(TapRecognizer, e => {
-
-        // A custom event is necessary here (as opposed to the click() function)
-        // because some targets (e.g. SVGs) may not be HTMLElements.
-        const event = createCustomEvent(
-            this.win,
-            'click',
-            null,
-            {bubbles: true}
-        );
-        e.data.target.dispatchEvent(event);
-      });
+      this.gestures_.onGesture(TapRecognizer, this.handleTap_.bind(this));
     }
   }
 
+  /**
+   * @param {*} e
```

Comment:
It can help to prefer a specific type over `*`. Is it `Gesture<DoubletapDef>`?

Reply:
So technically, the types for the signals emitted by the `Gestures` are `SwipeDef`, `DoubletapDef`, `TapDef`, whose types would look something of the form: 

```
   * @param {!../../../src/gesture-recognizers.SwipeDef} e
```


 or DATA from 

https://github.com/ampproject/amphtml/blob/45f4f7d7e4017190ac09bbac88fdae5764ec006a/src/gesture.js#L563. 

```
   * @param {!../../../src/gesture.DATA} e
```

However, the type passed in is actually * from https://github.com/ampproject/amphtml/blob/45f4f7d7e4017190ac09bbac88fdae5764ec006a/src/gesture.js#L395

So I don't think we can type it to anything other than * unless there's an upstream change.

Reply:
I see, thanks for explaining